core/kazoo_convert/src/kz_fax_converter.erl

```diff
@@ -0,0 +1,282 @@
+                     {'ok', any()}|{integer(), non_neg_integer()}|{'error', any()}.
+convert(From, To, Content, Opts) ->
+    Options = maps:from_list(
+                [{<<"from_format">>, From}
+                ,{<<"to_format">>, To}
+                ,{<<"job_id">>, props:get_value(<<"job_id">>, Opts, kz_binary:rand_hex(12))}
+                 | props:delete_keys([<<"job_id">>], Opts)
+                ]),
+    Filename = save_file(Content, Options),
+    lager:info("converting document ~s from ~s to ~s", [Filename, From, To]),
+    case run_convert(eval_format(From, To), To, Filename, Options) of
+        {'ok', _}=Ok ->
+            lager:info("succesfully converted file: ~s to format: ~s", [Filename, To]),
+            Ok;
+        {'ok', _, _}=Ok ->
+            lager:info("succesfully converted file: ~s to format: ~s", [Filename, To]),
+            Ok;
+        {'error', Message}=Error ->
+            lager:error("conversion failed with error: ~p", [Message]),
+            Error
+    end.
+
+-spec eval_format(kz_term:ne_binary(), kz_term:ne_binary()) -> kz_term:list().
+eval_format(<<"image/", _SubType/binary>>, ?TIFF_MIME) ->
+    [fun image_to_tiff/2
+    ];
+eval_format(Format, Format) ->
+    [];
+eval_format(?TIFF_MIME, ?PDF_MIME) ->
+    [fun tiff_to_pdf/2
+    ];
+eval_format(?PDF_MIME, ?TIFF_MIME) ->
+    [fun pdf_to_tiff/2
+    ];
+eval_format(<<?OPENXML_MIME_PREFIX, _/binary>>, ?TIFF_MIME) ->
+    [fun openoffice_to_pdf/2
+    ,fun pdf_to_tiff/2
+    ];
+eval_format(CT, ?TIFF_MIME) when ?OPENOFFICE_COMPATIBLE(CT) ->
+    [fun openoffice_to_pdf/2
+    ,fun pdf_to_tiff/2
+    ];
+eval_format(<<?OPENOFFICE_MIME_PREFIX, _/binary>>, ?TIFF_MIME) ->
+    [fun openoffice_to_pdf/2
+    ,fun pdf_to_tiff/2
+    ];
+eval_format(<<?OPENXML_MIME_PREFIX, _/binary>>, ?PDF_MIME) ->
+    [fun openoffice_to_pdf/2
+    ];
+eval_format(<<?OPENOFFICE_MIME_PREFIX, _/binary>>, ?PDF_MIME) ->
+    [fun openoffice_to_pdf/2
+    ];
+eval_format(CT, ?PDF_MIME) when ?OPENOFFICE_COMPATIBLE(CT) ->
+    [fun openoffice_to_pdf/2
+    ];
+eval_format(FromFormat, ToFormat) ->
+    {'error', <<"invalid conversion requested: ", FromFormat/binary, " to: ", ToFormat/binary>>}.
+
+-spec run_convert({atom(), kz_term:ne_binary()}, kz_term:ne_binary(), kz_term:ne_binary(), kz_term:proplist()) -> kz_term:list().
+run_convert({'error', _}=Error, _ToFormat, _FilePath, _Options) ->
+    Error;
+run_convert([Operation|Operations], ToFormat, FilePath, Options) ->
+    case Operation(FilePath, Options) of
+        {'ok', OutputPath} ->
+            maybe_delete_previous_file(FilePath, OutputPath),
+            run_convert(Operations, ToFormat, OutputPath, Options);
+        Error -> Error
+    end;
+run_convert([], ToFormat, FilePath, Options) ->
+    case validate_output(ToFormat, FilePath, Options) of
+        {'ok', _} ->
+            format_output(ToFormat, FilePath, Options);
```

Comment:
call this like `format_output(ToFormat, return_output(FilePath, Options), Options);` to decouple return out put format (binary or file path) and get sizes logic.